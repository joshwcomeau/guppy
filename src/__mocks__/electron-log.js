/* eslint-disable flowtype/require-valid-file-annotation */

module.exports = {
  transports: {
    file: {
      level: 'info',
    },
  },
};
