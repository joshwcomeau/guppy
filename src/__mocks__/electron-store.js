/* eslint-disable flowtype/require-valid-file-annotation */
module.exports = function() {
  let store = {};

  this.clear = jest.fn(() => {
    store = {};
  });

  this.get = jest.fn(key => store[key]);
  this.set = jest.fn((key, val) => {
    store[key] = val;
    return val;
  });
};
