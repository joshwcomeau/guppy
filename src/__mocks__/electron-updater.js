/* eslint-disable flowtype/require-valid-file-annotation */

module.exports = {
  autoUpdater: {
    checkForUpdatesAndNotify: jest.fn(),
    on: jest.fn(),
  },
};
