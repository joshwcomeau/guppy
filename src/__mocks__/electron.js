/* eslint-disable flowtype/require-valid-file-annotation */
import * as path from 'path';

module.exports = {
  ipcRenderer: {
    send: jest.fn(),
  },
  remote: {
    app: {
      getAppPath: () => path.resolve(__dirname, '..', '..', '..'),
      getPath: () =>
        process.env.APPDATA ||
        (process.platform === 'darwin'
          ? process.env.HOME + 'Library/Preferences'
          : '/var/local'),
    },
    dialog: {
      showErrorBox: jest.fn(),
      showOpenDialog: jest.fn(),
      showMessageBox: jest.fn(),
    },
    shell: {
      moveItemToTrash: jest.fn(),
    },
  },
};
