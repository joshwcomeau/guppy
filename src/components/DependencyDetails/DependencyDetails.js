// @flow
import React, { PureComponent, Fragment } from 'react';
import styled from 'styled-components';

import { COLORS } from '../../constants';

import Heading from '../Heading';
import Spacer from '../Spacer';
import Label from '../Label';
import Button from '../Button';
import PixelShifter from '../PixelShifter';
import BigClickableButton from '../BigClickableButton';
import DependencyDetailsTable from '../DependencyDetailsTable';

import type { Dependency } from '../../types';

type Props = {
  projectId: string,
  dependency: Dependency,
};

class DependencyDetails extends PureComponent<Props> {
  render() {
    const { projectId, dependency } = this.props;

    const tags =
      dependency.keywords &&
      dependency.keywords.map(keyword => <Tag key={keyword}>{keyword}</Tag>);

    return (
      <Fragment>
        <Header>
          <PixelShifter y={-4}>
            <HeaderText>
              <Name size="small">{dependency.name}</Name>
              <Description>{dependency.description}</Description>
            </HeaderText>
          </PixelShifter>
        </Header>

        <VersionsWrapper>
          <Col>
            <VersionLabel>Installed Version</VersionLabel>
            <VersionNum>16.4.0</VersionNum>
          </Col>

          <Col>
            <VersionLabel>Latest Version</VersionLabel>
            <VersionNum>16.4.1</VersionNum>
          </Col>

          <Col>
            <Button
              size="small"
              type="fill"
              color1={COLORS.green[700]}
              color2={COLORS.lightGreen[500]}
            >
              Update
            </Button>
          </Col>
        </VersionsWrapper>

        <MainContent>
          <DependencyDetailsTable
            projectId={projectId}
            dependency={dependency}
          />
        </MainContent>
      </Fragment>
    );
  }
}

const Header = styled.header`
  position: relative;
  background-image: linear-gradient(
    45deg,
    ${COLORS.blue[700]},
    ${COLORS.violet[700]}
  );
  border-radius: 8px 8px 0 0;
  color: ${COLORS.white};
`;

const HeaderText = styled.div`
  padding: 15px;
`;

const Svg = styled.svg`
  position: absolute;
  left: 0;
  right: 0;
  bottom: 0;
  display: block;
  overflow: visible;
`;

const Name = styled(Heading)`
  color: ${COLORS.white};
`;

const Description = styled.div`
  font-size: 20px;
  font-weight: 400;
  color: ${COLORS.gray[300]};
  -webkit-font-smoothing: antialiased;
`;

const Tag = styled.span`
  display: inline-block;
  padding: 4px 12px;
  border-radius: 6px;
  font-size: 15px;
  background: ${COLORS.gray[700]};
  color: ${COLORS.white};
`;

const VersionsWrapper = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  padding: 15px;
  background: ${COLORS.gray[100]};
  border-bottom: 1px solid rgba(0, 0, 0, 0.05);
`;

const Col = styled.div`
  width: 150px;
  text-align: center;
`;

const VersionLabel = styled(Label)`
  color: ${COLORS.gray[600]};
`;

const VersionNum = styled.div`
  margin-top: 3px;
  font-size: 22px;
  font-weight: 600;
  -webkit-font-smoothing: antialiased;
  line-height: 22px;
`;

const Buttons = styled.div`
  display: flex;
  align-items: center;
  margin: 10px 0;
`;

const Action = styled(Button)`
  width: 130px;
`;

const MainContent = styled.div`
  padding: 15px;
`;

export default DependencyDetails;
