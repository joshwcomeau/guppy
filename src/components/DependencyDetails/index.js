// @flow
export { default } from './DependencyDetails';
