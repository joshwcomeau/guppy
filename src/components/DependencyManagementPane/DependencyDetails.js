// @flow
import React, { PureComponent, Fragment } from 'react';
import styled from 'styled-components';

import { COLORS, RAW_COLORS } from '../../constants';

import Heading from '../Heading';
import PixelShifter from '../PixelShifter';
import DependencyDetailsTable from './DependencyDetailsTable';
import DependencyInfoFromNpm from './DependencyInfoFromNpm';
import DependencyUpdateRow from './DependencyUpdateRow';

import type { Dependency } from '../../types';
import type { NpmResult } from './DependencyInfoFromNpm';

type Props = {
  projectId: string,
  dependency: Dependency,
  isOnline: boolean,
};

class DependencyDetails extends PureComponent<Props> {
  render() {
    const { projectId, dependency, isOnline } = this.props;

    return (
      <DependencyInfoFromNpm packageName={dependency.name}>
        {({ name, latestVersion, lastUpdatedAt, isLoading }: NpmResult) => (
          <Fragment>
            <Header>
              <PixelShifter
                y={-4}
                reason="Optical symmetry between top and left edge of parent"
              >
                <HeaderText>
                  <Name size="small">{dependency.name}</Name>
                  <Description>{dependency.description}</Description>
                </HeaderText>
              </PixelShifter>
            </Header>
            <VersionsWrapper>
              <DependencyUpdateRow
                projectId={projectId}
                dependency={dependency}
                isLoadingNpmInfo={isLoading}
                latestVersion={latestVersion}
              />
            </VersionsWrapper>
            <MainContent>
              <DependencyDetailsTable
                projectId={projectId}
                dependency={dependency}
                lastUpdatedAt={lastUpdatedAt}
                isOnline={isOnline}
              />
            </MainContent>
          </Fragment>
        )}
      </DependencyInfoFromNpm>
    );
  }
}

const Header = styled.header`
  position: relative;
  background-image: linear-gradient(
    45deg,
    ${RAW_COLORS.blue[700]},
    ${RAW_COLORS.violet[700]}
  );
  border-radius: 8px 8px 0 0;
  color: ${COLORS.textOnBackground};
`;

const HeaderText = styled.div`
  padding: 15px;
`;

const Name = styled(Heading)`
  color: ${COLORS.textOnBackground};
`;

const Description = styled.div`
  font-size: 20px;
  font-weight: 400;
  color: ${RAW_COLORS.transparentWhite[300]};
  -webkit-font-smoothing: antialiased;
`;

const VersionsWrapper = styled.div`
  padding: 15px;
  height: 72px;
  background: ${RAW_COLORS.gray[100]};
  border-bottom: 1px solid rgba(0, 0, 0, 0.05);
`;

const MainContent = styled.div`
  padding: 15px;
`;

export default DependencyDetails;
