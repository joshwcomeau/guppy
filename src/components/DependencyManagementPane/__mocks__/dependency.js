// @flow
export const mockReactHit = {
  name: 'React',
  version: '16.0.0',
  license: 'MIT',
  modified: 1549828927372,
  humanDownloadsLast30Days: 1000000,
};
