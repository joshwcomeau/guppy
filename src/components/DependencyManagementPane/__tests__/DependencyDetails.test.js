import React from 'react';
import { shallow } from 'enzyme';
import lolex from 'lolex';

import DependencyDetails from '../DependencyDetails';
import { mockReactHit } from '../__mocks__/dependency';

describe('DependencyDetails component', () => {
  lolex.install();

  const projectId = 'a-project';
  const wrapper = shallow(
    <DependencyDetails projectId={projectId} dependency={mockReactHit} />
  );

  it('should render', () => {
    expect(
      wrapper.renderProp('children')({
        name: mockReactHit.name,
        latestVersion: '16.8.0',
        lastUpdatedAt: Date.now(),
        isLoading: false,
      })
    ).toMatchSnapshot();
  });
});
