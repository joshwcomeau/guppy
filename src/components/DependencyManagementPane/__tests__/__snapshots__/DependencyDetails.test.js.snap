// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DependencyDetails component should render 1`] = `
<Fragment>
  <styled.header>
    <PixelShifter
      reason="Optical symmetry between top and left edge of parent"
      y={-4}
    >
      <styled.div>
        <Styled(Heading)
          size="small"
        >
          React
        </Styled(Heading)>
        <styled.div />
      </styled.div>
    </PixelShifter>
  </styled.header>
  <styled.div>
    <Connect(DependencyUpdateRow)
      dependency={
        Object {
          "humanDownloadsLast30Days": 1000000,
          "license": "MIT",
          "modified": 1549828927372,
          "name": "React",
          "version": "16.0.0",
        }
      }
      isLoadingNpmInfo={false}
      latestVersion="16.8.0"
      projectId="a-project"
    />
  </styled.div>
  <styled.div>
    <DependencyDetailsTable
      dependency={
        Object {
          "humanDownloadsLast30Days": 1000000,
          "license": "MIT",
          "modified": 1549828927372,
          "name": "React",
          "version": "16.0.0",
        }
      }
      lastUpdatedAt={0}
      projectId="a-project"
    />
  </styled.div>
</Fragment>
`;
